src/idc/zephyr_idc.c

```diff
@@ -29,6 +29,9 @@
 #include <rtos/alloc.h>
 #include <rtos/spinlock.h>
 #include <ipc/topology.h>
+#if defined(CONFIG_ARCH_XTENSA) && !defined(CONFIG_KERNEL_COHERENCE)
+#include <zephyr/arch/xtensa/cache.h>
+#endif
```

Comment:
Hmm, if these are not defined, wouldn't the build anyways fail when z_xtensa_cache_flush() is not defined?

Reply:
do not remember why I did this, afaik this is the same in `zephyr/lib/alloc.c`

Reply:
@kv2019i do we need to fix ```zephyr/lib/alloc.c``` or is this correct ?